tests/integration/components/bs-alert-test.js

```diff
@@ -88,6 +89,36 @@ module('Integration | Component | bs-alert', function (hooks) {
     assert.ok(action.calledOnce, 'onDismissed has been called.');
   });
 
+  test('When fade is true, onDismissed is called once when visible changes from true to false', async function (assert) {
+    let action = sinon.spy();
+    this.actions.dismissed = action;
+    this.isAlertShown = true;
+
+    await render(
+      hbs`<BsAlert @fade={{true}} @onDismissed={{action "dismissed"}} @visible={{this.isAlertShown}}>Test</BsAlert>`
+    );
+
+    set(this, 'isAlertShown', false);
+    await settled();
```

Comment:
I think these lines of code mimick clicking `<button>hide</button>` to trigger `this.hideAlert` in @jelhan's [bug reproduction](https://github.com/kaliber5/ember-bootstrap/issues/1264#issue-710100120)

Reply:
Is this test indeed failing without your guard?

Reply:
I added the tests first to check that they fail. When I added the guard, I saw that these tests passed. I haven't verified if adding the guard works by linking the addon to a new app, however.